src/chat.rs

```diff
@@ -1726,25 +1726,33 @@ impl Chat {
 
     /// Returns profile image path for the chat.
     pub async fn get_profile_image(&self, context: &Context) -> Result<Option<PathBuf>> {
-        if let Some(image_rel) = self.param.get(Param::ProfileImage) {
-            if !image_rel.is_empty() {
-                return Ok(Some(get_abs_path(context, Path::new(&image_rel))));
-            }
-        } else if self.id.is_archived_link() {
-            if let Ok(image_rel) = get_archive_icon(context).await {
-                return Ok(Some(get_abs_path(context, Path::new(&image_rel))));
-            }
+        if self.id.is_archived_link() {
+            // This is not a real chat, but the "Archive" button
+            // that is shown at the top of the chats list
+            let image_rel = get_archive_icon(context).await?;
+            return Ok(Some(get_abs_path(context, Path::new(&image_rel))));
         } else if self.typ == Chattype::Single {
+            // For 1:1 chats, we always use the same avatar as for the contact
+            // This is before the `self.is_encrypted()` check, because that function
+            // has two database calls, i.e. it's slow
             let contacts = get_chat_contacts(context, self.id).await?;
             if let Some(contact_id) = contacts.first() {
-                if let Ok(contact) = Contact::get_by_id(context, *contact_id).await {
-                    return contact.get_profile_image(context).await;
-                }
+                let contact = Contact::get_by_id(context, *contact_id).await?;
+                return contact.get_profile_image(context).await;
             }
-        } else if self.typ == Chattype::Broadcast {
-            if let Ok(image_rel) = get_broadcast_icon(context).await {
+        } else if !self.is_encrypted(context).await? {
+            // This is an email-contact chat, show a special avatar that marks it as such
```

Comment:
`mimeparser` supports setting group avatars even for unencrypted chats. Should we really unsupport this? Maybe it's still better to have that "email" character at the end of the chat/contact name, but add it artificially in the core (i agree that UIs shouldn't care about this)? Green verification checkmarks are shown at the end of the name as well, would be good to unify everything somehow. One more reason: let's say i have email contacts "Alice" and "Bob" which both don't have avatars. Currently they have different colors with "A" and "B" inside which help me distinguish them, but the PR suggests they look the same.

Reply:
Could you remind me why we need this email pictogram in the UIs at all? In release versions we don't have it. And if we need it, maybe it's still possible to preserve colors and put the email unicode char / emoji inside? This gray icon looks boring, e.g. i have many email threads and i'm glad to see all this spectrum. Btw, green checkmarks also clutter the UI, e.g. in WhatsApp i don't see anything like this. Just putting in one more opinion, it's fine to merge if that's already agreed